rest/src/main/kotlin/request/RequestHandler.kt

```diff
@@ -16,4 +20,13 @@ public interface RequestHandler {
      */
     @Throws(RestRequestException::class)
     public suspend fun <B : Any, R> handle(request: Request<B, R>): R
+
+    public suspend fun <T> intercept(builder: RequestBuilder<T>): RequestBuilder<T>
+}
+
+public fun  <T> RequestBuilder<T>.defaultInterception(route: Route<T>, token: String): RequestBuilder<T> {
+        if (route.requiresAuthorizationHeader) {
+            unencodedHeader(HttpHeaders.Authorization, "Bot $token")
+        }
+    return this
 }
```

Comment:
```suggestion
    public suspend fun <T> RequestBuilder<T>.intercept(): RequestBuilder<T> {
        defaultInterception(token)
    }
}

public fun  <T> RequestBuilder<T>.defaultInterception(token: String): RequestBuilder<T> = apply {
        if (route.requiresAuthorizationHeader) {
            unencodedHeader(HttpHeaders.Authorization, "Bot $token")
        }
}
```
Why not use a member extension function?
We can provide a default, `RequestBuilder` also has access to the route.